### PR TITLE
Switch from podman and hostpath for MariaDB adoption

### PR DESCRIPTION
**Pull request by holser**
Switch from podman to oc run, in order to access the clusterIP service network. This also solves connectivity problems when deploying openstack CR that uses isolnet, and podman fails accessing IPs on internalAPI service network. Use storageclass (defaults to crc-csi-hostpath-provisioner) insetad of Instead of local host directories.

### CODE REVIEW

File: docs_user/modules/openstack-mariadb_copy.adoc [hunk 1]
```diff
@@ -55,6 +55,7 @@ endif::[]
 ifeval::["{build}" == "downstream"]
 MARIADB_IMAGE=registry.redhat.io/rhosp-dev-preview/openstack-mariadb-rhel9:18.0
 endif::[]
+STORAGE_CLASS_NAME=crc-csi-hostpath-provisioner
```
Comment (marios) [inline docs_user/modules/openstack-mariadb_copy.adoc:58]: ci issue seems relevant 

```
STORAGE_CLASS_NAME: unbound variable
```

https://logserver.rdoproject.org/16/316/01956cc0cd0432f59da6ca9dfa2c40e9c322a285/github-check/data-plane-adoption-osp-17-to-extracted-crc/7790de8/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_with_ceph_out_2024-03-01T07:44:20EST.log

File: tests/roles/mariadb_copy/templates/dump_dbs.bash [hunk 2]
```diff
@@ -6,7 +6,52 @@
...
 
-podman run -i --rm --userns=keep-id -u $UID -v $PWD:$PWD:z,rw -w $PWD $MARIADB_IMAGE bash <<EOF
+oc apply -f - <<EOF
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: mariadb-data
+spec:
+  storageClassName: $STORAGE_CLASS_NAME
```
Comment (jistr) [inline tests/roles/mariadb_copy/templates/dump_dbs.bash:16]: Looking at the CI failure, this needs to go among `mariadb_copy_shell_vars_src`?

File: docs_user/modules/openstack-mariadb_copy.adoc [hunk 3]
```diff
@@ -55,16 +50,77 @@ endif::[]
 ifeval::["{build}" == "downstream"]
 MARIADB_IMAGE=registry.redhat.io/rhosp-dev-preview/openstack-mariadb-rhel9:18.0
 endif::[]
+STORAGE_CLASS=crc-csi-hostpath-provisioner
 # Replace with your environment's MariaDB Galera cluster VIP and backend IPs:
 SOURCE_MARIADB_IP=192.168.122.99
 declare -A SOURCE_GALERA_MEMBERS
-SOURCE_GALERA_MEMBERS=(
+SOURCE_GALERA_MEMBERS=
```
Comment (bogdando) [inline docs_user/modules/openstack-mariadb_copy.adoc:57]: lost ( character?

File: docs_user/modules/openstack-mariadb_copy.adoc [hunk 4]
```diff
@@ -55,16 +50,77 @@ endif::[]
...
+  resources:
+    requests:
+      storage: 10Gi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mariadb-copy-data
+  annotations:
+    openshift.io/scc: anyuid
```
Comment (bogdando) [inline docs_user/modules/openstack-mariadb_copy.adoc:93]: please add ```k8s.v1.cni.cncf.io/networks: internalapi```, which probably becomes```k8s.v1.cni.cncf.io/networks: internal_api``` after https://github.com/openstack-k8s-operators/install_yamls/pull/758

File: docs_user/modules/openstack-mariadb_copy.adoc [hunk 5]
```diff
@@ -73,18 +129,17 @@ SOURCE_DB_ROOT_PASSWORD=$(cat ~/tripleo-standalone-passwords.yaml | grep ' Mysql
 ----
 for i in "${!SOURCE_GALERA_MEMBERS[@]}"; do
   echo "Checking for the database node $i WSREP status Synced"
-  sudo podman run -i --rm --userns=keep-id -u $UID $MARIADB_IMAGE mysql \
+  oc exec mariadb-copy-data -- bash -c 'mysql \
```
Comment (bogdando) [inline docs_user/modules/openstack-mariadb_copy.adoc:132]: please use " so vat env vars got interpolated

File: docs_user/modules/openstack-mariadb_copy.adoc [hunk 6]
```diff
@@ -238,4 +285,10 @@ diff -Z <(echo $novadb_svc_records) <(echo $PULL_OPENSTACK_CONFIGURATION_NOVA_CO
 related to the `restricted:latest` security context constraint. This is due to default security
 context constraints and will not prevent pod creation by the admission controller. You'll see a
 warning for the short-lived pod but it will not interfere with functionality.
+* Delete the ovn-data pod and persistent volume claim with OVN databases backup (consider making a snapshot of it, before deleting)
```
Comment (bogdando) [inline docs_user/modules/openstack-mariadb_copy.adoc:288]: mariadb-data

File: tests/roles/mariadb_copy/tasks/main.yaml [hunk 7]
```diff
@@ -6,16 +6,78 @@
...
+      resources:
+        requests:
+          storage: 10Gi
+    ---
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: mariadb-copy-data
+      annotations:
+        openshift.io/scc: anyuid
```
Comment (bogdando) [inline tests/roles/mariadb_copy/tasks/main.yaml:34]: the same https://github.com/openstack-k8s-operators/data-plane-adoption/pull/316/files#r1511181345

**Comment by bogdando** [comment]
> access the clusterIP service network

this looks like no longer accurate as we need it connected to internal_api isolnet, @hjensas wdyt?

**Review by bogdando — CHANGES_REQUESTED** [review]
well done, please see me comments

File: tests/roles/mariadb_copy/templates/dump_dbs.bash [hunk 8]
```diff
@@ -3,17 +3,14 @@
...
-    echo "Dumping \${dbname}"
-    mysqldump -h $SOURCE_MARIADB_IP -uroot "-p$SOURCE_DB_ROOT_PASSWORD" \
-        --single-transaction --complete-insert --skip-lock-tables --lock-tables=0 \
-        "\${dbname}" > "\${dbname}".sql
-done
-
-EOF
\ No newline at end of file
+# Note Filter the information and performance schema tables
+# Gnocchi is no longer used as a metric store, skip dumping gnocchi database as well
```
Comment (bogdando) [inline tests/roles/mariadb_copy/templates/dump_dbs.bash:7]: would be nice to note this extra change in the commit msg

File: docs_user/modules/openstack-mariadb_copy.adoc (hunk 5 above)
Comment (bogdando) [reply docs_user/modules/openstack-mariadb_copy.adoc:132]: or you can as well use rsh https://github.com/openstack-k8s-operators/data-plane-adoption/pull/316/files#diff-d5dba85fef50c60b9a5a0ead4321eea3518b41e38cb65892376198a8cd23c1c8R77 ?

**Comment by hjensas** [comment]
> > access the clusterIP service network
> 
> this looks like no longer accurate as we need it connected to internal_api isolnet, @hjensas wdyt?

I think we are in a bad situation between the two repos, we need update both and both run the same test job ...

The standalone is currently deploying mariadb binding to the ctlplane network. So we need this initially to either attach to ctlplane or not attach any network? Then do follow ups after getting the network isolation corrected in install_yamls standalone config.

I would suggest getting this done, and then do changes to clear out the network isolation thing separately?

**Comment by jistr** [comment]
It looks good! I wonder if we want to tackle the documentation changes in a separate PR since we got the green CI here? If the other PR only edits docs then it won't trigger the big job.

File: docs_user/modules/openstack-mariadb_copy.adoc (hunk 1 above)
Comment (holser) [reply docs_user/modules/openstack-mariadb_copy.adoc]: @marios All CI issues are addressed in latest PR.

File: docs_user/modules/openstack-mariadb_copy.adoc (hunk 3 above)
Comment (holser) [reply docs_user/modules/openstack-mariadb_copy.adoc]: Yes. Thank you. Parentheses moved back

File: docs_user/modules/openstack-mariadb_copy.adoc (hunk 5 above)
Comment (holser) [reply docs_user/modules/openstack-mariadb_copy.adoc:132]: Doc changes will be reflected in next PR

File: docs_user/modules/openstack-mariadb_copy.adoc [hunk 9]
```diff
@@ -55,6 +50,7 @@ endif::[]
 ifeval::["{build}" == "downstream"]
 MARIADB_IMAGE=registry.redhat.io/rhosp-dev-preview/openstack-mariadb-rhel9:18.0
 endif::[]
+STORAGE_CLASS=crc-csi-hostpath-provisioner
```
Comment (odyssey4me) [inline docs_user/modules/openstack-mariadb_copy.adoc:53]: This is an odd addition - is this appropriate?

Comment (holser) [reply docs_user/modules/openstack-mariadb_copy.adoc:53]: Yes, it does. The same pattern is shown at https://github.com/openstack-k8s-operators/data-plane-adoption/blob/main/tests/roles/ovn_adoption/tasks/main.yaml#L57 which is defined at https://github.com/openstack-k8s-operators/data-plane-adoption/blob/main/tests/roles/ovn_adoption/defaults/main.yaml#L2

Comment (jistr) [reply docs_user/modules/openstack-mariadb_copy.adoc:53]: I think the value is CRC-specific and so they wouldn't apply to customers verbatim. We do say "The values are just illustrative, use values that are correct for your environment". Pedro used "local-storage" which is probably a better default for downstream.

Comment (jistr) [reply docs_user/modules/openstack-mariadb_copy.adoc:53]: If you could post a separate PR with the docs fixes (on top of this PR content essentially), we could land this one without having to re-run CI i think.

File: docs_user/modules/openstack-mariadb_copy.adoc (hunk 4 above)
Comment (bogdando) [reply docs_user/modules/openstack-mariadb_copy.adoc:93]: @holser I think we need this, please see the same for ovn data copy tasks

File: tests/roles/mariadb_copy/tasks/main.yaml (hunk 7 above)
Comment (bogdando) [reply tests/roles/mariadb_copy/tasks/main.yaml:34]: nvm, I agree with "I would suggest getting this done, and then do changes to clear out the network isolation thing separately?"

File: docs_user/modules/openstack-mariadb_copy.adoc (hunk 4 above)
Comment (bogdando) [reply docs_user/modules/openstack-mariadb_copy.adoc:93]: nvm, agreed with Harald

**Review by bogdando — APPROVED** [review]

**Pull request merged**